kv_test.go

```diff
@@ -251,3 +251,51 @@ func TestKeyValueCreate(t *testing.T) {
 		t.Fatalf("Unexpected error code, got: %v", kerr.APIError().ErrorCode)
 	}
 }
+
+func TestKeyValueSourcing(t *testing.T) {
+	s := RunBasicJetStreamServer()
+	defer shutdownJSServerAndRemoveStorage(t, s)
+
+	nc, js := jsClient(t, s)
+	defer nc.Close()
+
+	kvA, err := js.CreateKeyValue(&KeyValueConfig{Bucket: "A"})
+	if err != nil {
+		t.Fatalf("Error creating kv: %v", err)
+	}
+
+	_, err = kvA.Create("keyA", []byte("1"))
+	if err != nil {
+		t.Fatalf("Error creating key: %v", err)
+	}
+
+	if _, err := kvA.Get("keyA"); err != nil {
+		t.Fatalf("Got error getting keyA from A: %v", err)
+	}
+
+	kvB, err := js.CreateKeyValue(&KeyValueConfig{Bucket: "B"})
+	if err != nil {
+		t.Fatalf("Error creating kv: %v", err)
+	}
+
+	_, err = kvB.Create("keyB", []byte("1"))
+	if err != nil {
+		t.Fatalf("Error creating key: %v", err)
+	}
+
+	kvC, err := js.CreateKeyValue(&KeyValueConfig{Bucket: "C", Sources: []*StreamSource{{Name: "A"}, {Name: "B"}}})
+	if err != nil {
+		t.Fatalf("Error creating kv: %v", err)
+	}
+
+	// Wait half a second to make sure it has time to populate the stream from it's sources
+	time.Sleep(500 * time.Millisecond)
```

Comment:
Always waiting 500ms in a test is kind of a lot... It would be good to either decrease this (if possible) or at least replace this with some function which periodically checks for the result (with a timeout), something like `waitFor()` func in drain_test.go

Reply:
Reduced to 100 ms